docs/Advanced/SQL Searches/SQL Columns.md

```diff
@@ -0,0 +1,56 @@
+---
+publish: true
+---
+
+# Columns / Properties
+
+<span class="related-pages">#advanced/sql-search</span>
+
+The following columns or properties are available to be used in the WHERE clauses when writing the SQL queries. The SQL engine also allows you to access the columns or properties children. For the remainder of this document the term column will be used to reference a property of the task that may or may not have sub properties you can access
+
+## Available Columns
+
+| Column Name              | Description                                                                         | Type                |
+| ------------------------ | ----------------------------------------------------------------------------------- | ------------------- |
+| status                   | The status object holds the current and next indicator and a description.           | Status Object       |
+| status->symbol           | The value between the square brackets. ('x', ' ', '-', '/', etc.)                   | string              |
+| status->name             | The display name for the status. ('Done', 'Todo', 'Cancelled', 'In Progress')       | string              |
+| status->nextStatusSymbol | The next indicator to be used when clicked on.  ('x', ' ', '/', etc.)               | string              |
+| description              | The description of the task.                                                        | string              |
+| indentation              | The indentation type used like \*, -, > -. etc.                                     | string              |
+| listMarker               | The list marker type, -, \* or NUMBER.                                              | string              |
+| taskLocation             | Location object holds the below properties                                          | TaskLocation Object |
+| taskLocation->lineNumber | Line number the task is on in the note.                                             | string              |
+| precedingHeader          | The heading that precedes the task.                                                 | string              |
+| path                     | Path to the note that contains the task.                                            | string              |
```

Comment:
Feedback from pshev on [Discord](https://discord.com/channels/686053708261228577/965681451297304596/1097212035114221668):

- you need to document the way to exclude a folder and subfolders.  for example,  when I did the above,

AND path NOT LIKE 'docs/%'

worked  but:

AND path NOT LIKE '/docs/%'

gave me items in the docs directory, but not all.

Reply:
The path property is a string like "1 Journal/2023/2023-04/2023-04-17.md"

If you want to exclude all the files in the journal folder then `path NOT LIKE '1 Journal/%'` will exclude app paths with '1 Journal/' at the start of the string.

From your query it looks like you want to exclude files if that value is somewhere in the path. So in this case it would be `path NOT LIKE '%/2023/%'` which would filter out all the tasks where their path contains 2023, this would also include folders like "9 datastore/habits/2023/04" in it so any tasks on pages with 2023 in the path in '9 datastore' would also be excluded.

It is less about excluding folders and sub folders and more about treating the path as a string and then matching on it and using the wild cards correctly for LIKE.

Will add something to the WHERE page.

Reply:
Updated the WHERE page with more details on like.